js/ui/tab_panel.js

```diff
@@ -22,6 +22,8 @@ const TABPANEL_CONTAINER_CLASS = 'dx-tabpanel-container';
 
 const TABS_ITEM_TEXT_CLASS = 'dx-tab-text';
 
+const FOCUS_ON_DISABLED_CLASS = 'dx-focus-on-disabled';
```

Comment:
This class seems to be really strange to me. Cannot we just use `.dx-state-focused.dx-state-disabled` selector instead of this?

Reply:
No, this class is being used for setting box-shadow properties to another items like navigation buttons, tabs and another tab-elements, when disabled item has focused state. There is no other way to determine this.

Please see _index.scss file for tabPanel: [[link](https://github.com/marker-dao/DevExtreme/blob/23_1_disabled_tab_panel_issues/scss/widgets/generic/tabPanel/_index.scss#L120)].

Reply:
Let's discuss it in voice

Reply:
Okay, let's rename it to `dx-disabled-tab-focused`

Reply:
Updated class name here [10a76f9d9152aea3874d807cb4712c252ec11d44].